File: front_endf/lib/models/comment.dart
import 'package:blogapp/models/agence.dart';

class Comment {
  int? id;
  String? comment;
  String? name;

  Comment({this.id, this.comment, this.name});

  // map json to comment model
  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'],
      comment: json['comment'],
      name: json['name'],
    );
  }
}

File: back_endl/app/Models/Comment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\User;

class Comment extends Model
{
    use HasFactory;

    protected $fillable = [
        'comment',
        'name',
        'user_id',
        'offer_id'
    ];

    public function user()
    {
        return $this->belongsTo(Agence::class);
        //return $this->belongsTo(client::class);
        
        
    }
}
